enhancements/microshift/gateway-api-support.md

```diff
@@ -0,0 +1,434 @@
+ingress routing capabilities, also aligned with the way Kubernetes foresees
+this functionality.
+
+### User Stories
+* As a MicroShift admin, I want to add different Gateway resources to the
+  cluster for applications to create routes on them.
+* As a MicroShift application developer, I want to add HTTP routes to specific
+  Gateways so that I can express routing capabilities using an upstream API.
+* As a MicroShift application developer, I want to add GRPC routes to specific
+  Gateways so that I can express routing capabilities using an upstream API.
+
+### Goals
+- Full support for core, non-experimental resources, including gateways, HTTP routes and GRPC routes.
+- Full support for multiple Gateway instances.
+- Full support for north/south [use case](https://gateway-api.sigs.k8s.io/concepts/use-cases/#basic-northsouth-use-case).
+- Lightweight implementation, consuming as little resources as possible.
+
+### Non-Goals
+- Automatically removing Gateway API from the cluster upon RPM uninstall
+- Provide support for experimental resources, such as those listed [here](https://gateway-api.sigs.k8s.io/concepts/api-overview/#route-resources).
+- Support different GatewayClasses with the same controller.
+
+## Proposal
+As the Gateway API is still in development and MicroShift already provides
+means for routing capabilities, this feature should be optional.
+
+As with other optional features there will be an optional RPM with all the
+required manifests to be applied upon MicroShift's start. The backing
+implementation for Gateway API support will be based on the latest OpenShift
+Service Mesh component, which is an optional operator from OpenShift.
+
+OpenShift Service Mesh (OSSM) is the service mesh operator from OpenShift.
+Since version 3 (which is going to be Tech Preview at the time of this writing)
+it is based in the [sail operator](https://github.com/openshift-service-mesh/sail-operator/tree/main),
+and is released as an optional operator in the Red Hat operators catalog for
+OLM. This operator deploys all of Istio's manifests and is capable of handling
+Istio control plane deployments, Istio CNIs and other components. Istio has
+first class support for Gateway API resources and is already supported by Red
+Hat.
+
+### Workflow Description
+**User** is a human user responsible for setting up and managing Edge Devices.
+**Application** is user's workload that intends to use routing capabilities.
+
+#### Installation and usage on RHEL For Edge (ostree)
+> In this workflow, it doesn't matter if the device is already running R4E with existing MicroShift cluster.
+> Deployment of new commit requires reboot which will force recreation of the Pod networking after adding Multus.
+
+1. User gathers all information about the networking environment of the edge device.
+1. User prepares ostree commit that contains:
+   - MicroShift RPMs.
+   - Gateway API for MicroShift RPM.
```

Comment:
Will "Gateway API for MicroShift" RPM be dependent on microshift-multus RPM?

Reply:
This was a typo.